### ISSUE TITLE
An Inquiry only 

### ISSUE THREAD

**Issue by Gamedevolop** [vbohush/SortingAlgorithmAnimations#2]
Hi Hello, 

Thanks very much for all the example are really helpful
I don't know Java but looking into the code I could pick up a few things, the one thing I'm interested in that I couldn't figure out is what colours values are you using? RGB?  what is the logic that you use to sort them?

Would you be so kind as to explain to me the process as I been looking to do colour sorting for a week but all the Algorithms that I tested from StackOverflow it didn't work for me?
I'm referring exactly to the example >>>>QuickSortPanel.java

I would love to do the same but in JS if I could understand the logic behind it

Thank you very much

**Comment by vbohush** [comment]
Hello!
Please search source code for **setColor** to see what colors I'm using, for exaple [here](https://github.com/vbohush/SortingAlgorithmAnimations/blob/master/src/net/bohush/sorting/QuickSortPanel.java#L103). I am using 9 sorting algorithms as described in [readme](https://github.com/vbohush/SortingAlgorithmAnimations/blob/master/README.md). Just google it if you want to know how exactly they work. There are tons of articles about each of them, which can explain far more better than I.

**Comment by Gamedevolop** [comment]
> Hello!
> Please search source code for **setColor** to see what colors I'm using, for exaple [here](https://github.com/vbohush/SortingAlgorithmAnimations/blob/master/src/net/bohush/sorting/QuickSortPanel.java#L103). I am using 9 sorting algorithms as described in [readme](https://github.com/vbohush/SortingAlgorithmAnimations/blob/master/README.md). Just google it if you want to know how exactly they work. There are tons of articles about each of them, which can explain far more better than I.



Hi vbohush

Thank Very much you for your replay

I looked at that code but I'm still having trouble to understand Wich value do you use to sort the colours? not to create the colours

I have been looking on Google how it works the "QuickSortPanel" Algorithms and I have already an idea of how it works but the only thing that I'm missing its wich values do you use for sorting the Colors specifically, are you using the Colors RGB? or something else?

Can you explain a bit more just the part of how the colours get organized where do you get the values what is the logic to sort them out? 

Example:
If I want to sort numbers instead of colours (9, 8, 6, 3, 2, 1, 7, 4) 
that Logycly it will be sorted by index number from low to high 
(1,2,3,4,6,7,8,9)

But the colours how they get sorted do you assign any numbers to them? or how? 

Can you please just help me on that, as its the only thing that confuses me

Thank you very much for your time

**Comment by Gamedevolop** [comment]
To be more clear I know the explanation of how the algorithm works as you can see on this tutorial [quicksort Tutorial ](https://www.guru99.com/quicksort-in-javascript.html)

but I still don't know how this can be done with colours? 
With just normal Numbers is easy but with colours which values you are assigning to them so you can use it to sort them out?

**Comment by Gamedevolop** [comment]
No worries I found already the solution with the help of someone.

The Answer I was looking for is:

-For the Values, you are Using "HSB" 
-And you are Targetting just the "H" 

So you are sorting out from the lowest "H" to the Highest "H"

Thank you

**Comment by vbohush** [comment]
Yes, you're right, I am using HSB and sort only by H, from lowest to highest.
